### ISSUE TITLE
error when trying to change folder names

### ISSUE THREAD

**Issue by remisharrock** [France-ioi/task-editor#14]
there is an error when we modify folder names on git, push to the server and go back to task-editor :
https://github.com/remisharrock/c-programming-with-linux-MOOC-issues-tracker/issues/573

**Comment by mblockelet** [comment]
Hi, I restored the Git state to something which is usable for now.

From what I could reconstruct, the issue seems that a folder was both edited in the task-editor, and renamed on the github repostiory, leading to an inconsistent state where there were modifications on both sides and no easy way to merge them.

I manually restored to a consistent state, but I will have to see how to resolve automatically these kind of conflicts, as it wasn't planned in the task-editor that a folder could be renamed remotely while it is edited locally.

**Comment by petraBT** [comment]
Is it possible that the path to a task is stored somewhere internally and not updated when a task is manually moved or renamed in git? While we were able to open such a moved/renamed task in taskeditor, when attempting to save it from within taskeditor a folder was created (by taskeditor?) in the old location, the problem description (the html) saved inside it, and then things seems to go off the rails.

**Comment by mblockelet** [comment]
Hi,

After rereading the code, I don't see anywhere where the path to a task is stored internally.

I'm not sure what could have happened there : are you sure the task wasn't saved from a window opened onto the old path? The current path is only stored on the client side.

Also, do you have the reference of the commit where task-editor recreated the task? I'm interested in looking at what could have happened at that moment!

Thanks!

**Comment by petraBT** [comment]
I don’t know whether the task was accidentally left open elsewhere - there were four of us playing around at the same time from four different computers, so it may well have been the case unfortunately.

The task we were working on was “Comments” in v2/C101/Week1/

I am not familiar enough with git unfortunately to find the Commit that showed this state, but I still have that folder sitting in my trash on my local computer and am attaching it to this message. Taskeditor created two versions of the problem description inside this folder, one named index_fa.html, one named index_fr.html (and these appear to be exact copies of what is normal called index.html).

I hope this helps…

Thanks- Petra



On Nov 19, 2019, at 8:15 AM, mblockelet <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi,

After rereading the code, I don't see anywhere where the path to a task is stored internally.

I'm not sure what could have happened there : are you sure the task wasn't saved from a window opened onto the old path? The current path is only stored on the client side.

Also, do you have the reference of the commit where task-editor recreated the task? I'm interested in looking at what could have happened at that moment!

Thanks!

**Comment by mblockelet** [comment]
It seems the issue happened in the merge commit c0dd53e in that repository : multiple modifications at the same time.

About the files `index_fr.html ` and `index_fa.html`, they're currently automatically generated because of a bug that we didn't have time to fix. It won't affect anything in the meanwhile.

**Comment by remisharrock** [comment]
Thanks for investigating that.
Does it mean that, until there is a fix for that problem, we have to make sure the task we want to move is not opened anywhere with task-editor (if it is we save it and close task-editor) and that the first thing to do when re-opening task-editor is to refresh so that is pulls the modifications fot github ?

**Comment by mblockelet** [comment]
Yes, but you'll need to do it even after the fix is put in. The fix will mostly be to avoid ending up in an inconsistent state, in which I would need to manually sort things out ; this way, whatever happens, you won't get stuck in the task-editor or unable to save anything as it has been the issue there.

However, there is no good behavior that the task-editor can have if the repository is edited manually **at the same time** as someone is editing a task in the task-editor. Merging modifications on the same files is hard ; git itself didn't manage to do it automatically, which is why the issue happened in the first place, and the task-editor can't be smarter than git. So, all I can do with a fix, is ensure the local copy the task-editor uses doesn't get in a stuck state, and that if there's any conflicts, the task-editor still pushes its modifications, for you to solve manually on the repository.

So please avoid renaming a task while someone else is working on it ;)

**Comment by remisharrock** [comment]
Do we still have this issue with sapces in folder names ?
